src/module.c

```diff
@@ -3047,6 +3050,76 @@ int RM_HashGet(RedisModuleKey *key, int flags, ...) {
     return REDISMODULE_OK;
 }
 
+/* --------------------------------------------------------------------------
+ * Key API for the stream type.
+ * -------------------------------------------------------------------------- */
+
+/**
+ * Adds an entry to a stream. Like XADD without trimming.
+ *
+ * - `key`: The key where the stream is (or will be) stored
+ * - `id`: If the `AUTOID` flag is set, this is where the assigned ID is
+ *   returned. Can be NULL if `AUTOID` is set, if you don't care to receive the
+ *   ID. If `AUTOID` is not set, this is the requested ID.
+ * - `flags`: A bit field of
+ *   - `REDISMODULE_STREAM_AUTOID`: Assign a stream ID automatically, like `*`
+ *     in the XADD command.
+ *   - `REDISMODULE_STREAM_NOMKSTREAM`: Don't create the stream if it doesn't
+ *     exist.
+ * - `argv`: A pointer to an array of size `numfields * 2` containing the
+ *   fields and values.
+ * - `numfields`: The number of field-value pairs in `argv`.
+ *
+ * Returns `REDISMODULE_OK` if an entry has been added. Returns
+ * `REDISMODULE_ERR` if the key is not opened for writing, if the key has a type
+ * other than stream, if the key doesn't exist in case the flag `NOMKSTREAM` is
+ * set or when attempting to create an entry with an id which is not allowed.
+ */
+int RM_StreamAdd(RedisModuleKey *key, RedisModuleStreamID *id, int flags,
+                 RedisModuleString **argv, int64_t numfields) {
+    /* Validate args */
+    if (!(key->mode & REDISMODULE_WRITE))
+        return REDISMODULE_ERR;
+    if (key->value && key->value->type != OBJ_STREAM)
+        return REDISMODULE_ERR;
+    if (key->value == NULL && (flags & REDISMODULE_STREAM_NOMKSTREAM))
+        return REDISMODULE_ERR;
+    if (!(flags & REDISMODULE_STREAM_AUTOID) &&
+        (id == NULL || (id->ms == 0 && id->seq == 0)))
+        return REDISMODULE_ERR;
+
+    /* Create key if necessery */
+    if (key->value == NULL)
+        moduleCreateEmptyKey(key, REDISMODULE_KEYTYPE_STREAM);
+
+    stream *s = key->value->ptr;
+
+    /* We don't assume streamID and RedisModuleStreamID are identical.
+       Should we? */
```

Comment:
technically, i think it's ok.. they're identical and will never change.
but i can't see any preexisting practice like that, and since it's just to field, copying them one by one is easy and clean (which is what you did), so i guess that's a better choice.